DEVELOPERS.md

```diff
@@ -276,3 +282,38 @@ An intermediate step generates a JSON file (`public_docs.json`) containing the d
 To generate this file, run:
 
     just generate-docs
+
+### Making changes to the dataset definition snippets
+
+These snippets are separate from the tutorial examples in `databuilder/ehrql-tutorial-examples`.
+There is a separate README in that directory that explains how those tutorial examples work.
+We may eventually unify the tutorial examples with the snippet
+so that all example code is checked in the same way.
+
+Edit the python modules in the `databuilder/snippets` directory.
+
+Examples are included in the markdown files using the [pymdown snippet notation](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/#snippets-notation).
+
+Each of the snippets sections in each snippet Python source file are bounded by markers:
+
+```python
+# --8<-- [start:print]
+print("hello world")
+# --8<-- [end:print]
+```
+
+If this example was stored as `databuilder/snippets/hello.py`,
+then it could be included in the documentation Markdown source via:
+
+````
+```python
+;--8<-- 'databuilder/snippets/hello.py:print'
```

Comment:
Is this commented out on purpose?  I had some problems with the README embedded in the docs folder because the code blocks like this were being interpreted by mkdocs as actual snippets (that didn't exist) - is that the issue here too?

Reply:
I'm not quite sure why I did this. Maybe I'd been testing it in a MkDocs file? This specific file shouldn't ever be part of the MkDocs build, so we can probably remove the `;`.

Since it took me a few minutes to find what this does again: https://facelessuser.github.io/pymdown-extensions/extensions/snippets/#escaping-snippets-notation